src/v/pandaproxy/handlers.cc

```diff
@@ -192,6 +192,19 @@ create_consumer(server::request_t rq, server::reply_t rp) {
     auto req_data = ppj::rjson_parse(
       rq.req->content.data(), ppj::create_consumer_request_handler());
 
+    if (req_data.format != "binary") {
+        throw parse::error(
+          parse::error_code::invalid_param, "format must be binary");
+    }
+    if (req_data.auto_offset_reset != "earliest") {
+        throw parse::error(
+          parse::error_code::invalid_param, "auto.offset must be earliest");
+    }
+    if (req_data.auto_commit_enable != "false") {
```

Comment:
why isn't this a bool ? shouldn't json parsing give you a c++ type? why stringly type this?

Reply:
This is from #1154 - you're right, it's probably a little too strict, but this is a string for compatibility reasons.

Reply:
got it. i'm ok w/ this for now.